File: endpoint/templates/go/gin/endpoint.controller.go
package <%= nameLowerCase %>controller

import (
	_ "encoding/json"
	_ "<%= repoHostUrl %>/<%= userNameSpace %>/<%= appName %>/server/api/<%= feature %>/dao"
	_ <%= nameLowerCase %> "<%= repoHostUrl %>/<%= userNameSpace %>/<%= appName %>/server/api/<%= feature %>/model"
	"github.com/gin-gonic/gin"
	_ "io/ioutil"
	_ "net/http"
)

func GetAll(c *gin.Context) {
}

func GetById(c *gin.Context) {
}

func New(c *gin.Context) {
}

func Update(c *gin.Context) {
}

func Remove(c *gin.Context) {
}

File: endpoint/templates/node/koa/typescript/endpoint.controller.ts
import * as <%= name %>DAO from '../dao/<%= name %>-dao';

export default class <%= name %>Controller {
  *getAll() {
    try {
      let <%= nameLowerCase %>s = yield <%= name %>DAO.getAll();
      this.body = <%= nameLowerCase %>s;
      this.status = 200;
    } catch(e) {
      this.status = 400;
    }
  }

  *createNew() {
    let _<%= nameLowerCase %> = this.body;

    try {
      this.body = yield <%= name %>DAO.createNew(_<%= nameLowerCase %>);
      this.status = 201;
    } catch(e) {
      this.status = 400;
    }
  }

  *removeById() {
    let _id = this.params.id;

    try {
      yield <%= name %>DAO.removeById(_id);
      this.status = 200;
    } catch(e) {
      this.status = 400;
    }
  }
}
